base/ltxref.dtx

```diff
@@ -229,9 +235,43 @@
   \else
    \expandafter#2#1\null
   \fi}
-\def\ref#1{\expandafter\@setref\csname r@#1\endcsname\@firstoftwo{#1}}
-\def\pageref#1{\expandafter\@setref\csname r@#1\endcsname
-                                   \@secondoftwo{#1}}
+%    \end{macrocode}
+%    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2022/06/01}%
+%<latexrelease>                 {\ref}{Add starred reference commands}%
+%    \end{macrocode}
+%    \begin{macrocode}
+\def\@kernel@sref#1{\expandafter\@setref\csname r@#1\endcsname\@firstoftwo{#1}}
+\def\@kernel@spageref#1{\expandafter\@setref\csname r@#1\endcsname
+                 \@secondoftwo{#1}}
+\let\@kernel@ref\@kernel@sref
+\let\@kernel@pageref\@kernel@spageref
+\NewDocumentCommand\ref{s}
+   {\IfBooleanTF{#1}{\@kernel@sref}{\@kernel@ref}}
```

Comment:
Given that the kernel doesn't do anything special with the star and we don't want other packages to overwrite `\kernel@...` there doesn't seem to be a need to provided two different commands that are then made the same. I think one used by both branches is clearer and simpler.

Reply:
I wondered about that. If we move at some time hyperlinking to the kernel then we would need them, wouldn't we? And then the switch would be easier (e.g. one wouldn't have to adapt showkeys again). But I don't mind to change it for now.

Reply:
That's an argument in favor of leaving it as you implemented it. Ok.